src/shogun/kernel/ANOVAKernel.cpp

```diff
@@ -7,44 +7,36 @@
 
 #include <shogun/mathematics/Math.h>
 #include <shogun/kernel/ANOVAKernel.h>
-#include <shogun/mathematics/Math.h>
 
 using namespace shogun;
 
-ANOVAKernel::ANOVAKernel(): DotKernel(0), cardinality(1.0)
+ANOVAKernel::ANOVAKernel(): DotKernel(0)
 {
-	register_params();
+	SG_ADD(
+	    &m_cardinality, "cardinality", "Kernel cardinality.",
+	    ParameterProperties::HYPER);
 }
 
-ANOVAKernel::ANOVAKernel(int32_t cache, int32_t d)
-: DotKernel(cache), cardinality(d)
+ANOVAKernel::ANOVAKernel(int32_t size, int32_t d)
+: ANOVAKernel()
 {
-	register_params();
+	set_cache_size(size);
+	m_cardinality = d;
 }
 
 ANOVAKernel::ANOVAKernel(
 	const std::shared_ptr<DenseFeatures<float64_t>>& l, const std::shared_ptr<DenseFeatures<float64_t>>& r, int32_t d, int32_t cache)
-  : DotKernel(cache), cardinality(d)
+  : ANOVAKernel(cache,d)
 {
-	register_params();
-	init(l, r);
+	ASSERT(DotKernel::init(l, r));
```

Comment:
I am curious, why is that changed?

Reply:
Just removed init(l,r) and pasted its code here. Also DotKernel::init(l,r) returns a bool, so I asserted that.